databroker/headersource/core.py

```diff
@@ -515,7 +515,12 @@ def insert_run_start(run_start_col, run_start_cache,
     col = run_start_col
     run_start = dict(time=time, uid=uid, **kwargs)
     apply_to_dict_recursively(run_start, sanitize_np)
-
+    try:
+        if run_start_given_uid(uid, run_start_col,
```

Comment:
If the RunStart with this uid already exists in the cache, no network call would be necessary. This is a very likely scenario if uids are generated by some user script. In terms of Mongo, the network call we are worrying about here is literally a few milli seconds since uids are already indexed. Are we that sensitive to latency? In case of sqlite, hdf5 etc. backends this is a call to some database/file loaded in memory.